File: mobile/lib/constants/tipo_casa_aluno_capacitacao.dart
/// Tipo de moradia do aluno de capacitação (valores da API).
abstract final class TipoCasaAlunoCapacitacao {
  static const propria = 'PROPRIA';
  static const cedida = 'CEDIDA';
  static const aluguel = 'ALUGUEL';

  static const opcoes = <String, String>{
    propria: 'Própria',
    cedida: 'Cedida',
    aluguel: 'Aluguel',
  };

  static String? rotulo(String? valor) {
    if (valor == null || valor.isEmpty) return null;
    return opcoes[valor];
  }
}

File: backend/src/lib/tipo-casa-aluno-capacitacao.ts
import { TipoCasaAlunoCapacitacao } from "@prisma/client";

export const TIPOS_CASA_ALUNO_CAPACITACAO = [
  { valor: "PROPRIA" as const, rotulo: "Própria" },
  { valor: "CEDIDA" as const, rotulo: "Cedida" },
  { valor: "ALUGUEL" as const, rotulo: "Aluguel" },
] as const;

export const TIPO_CASA_ALUNO_CAPACITACAO_VALUES =
  TIPOS_CASA_ALUNO_CAPACITACAO.map((e) => e.valor) as [
    TipoCasaAlunoCapacitacao,
    ...TipoCasaAlunoCapacitacao[],
  ];

const porValor = new Map(TIPOS_CASA_ALUNO_CAPACITACAO.map((e) => [e.valor, e]));

export function rotuloTipoCasaAlunoCapacitacao(
  valor: TipoCasaAlunoCapacitacao,
): string {
  return porValor.get(valor)?.rotulo ?? valor;
}
